packages/material-ui/src/Menu/Menu.js

```diff
@@ -126,16 +200,70 @@ const Menu = React.forwardRef(function Menu(props, ref) {
       });
     }
 
+    // If the current item in this map has nestedItems,
+    // we need the Menu to orchestrate its nestedMenu
+    if (hasNestedMenu && parentMenuOpen) {
```

Comment:
How do we iterate over children if the parent menu isn't open? Is that check necessary?